pkg/apis/autoscaling/types.go

```diff
@@ -95,6 +95,66 @@ type HorizontalPodAutoscalerSpec struct {
 	// more information about how each type of metric must respond.
 	// +optional
 	Metrics []MetricSpec
+
+	// Behavior contains settings to specify the scaling behavior both while scaling up
+	// and scaling down. It also is used to customize other aspects of scaling like the
+	// scale down stabilization window
+	// +option
+	Behavior *HPAScalingBehavior
+}
+
+// HPAScalingBehavior configures a scaling behavior for Up and Down direction
+// (scaleUp and scaleDown fields respectively)
+type HPAScalingBehavior struct {
+	// The scaling behavior while scaling up. Contains a list of policies and the policy selector.
+	// +optional
+	ScaleUp *HPAScalingDirectionBehavior
+	// The scaling behavior while scaling down. Contains a list of policies and the policy selector.
+	// +optional
+	ScaleDown *HPAScalingDirectionBehavior
+}
```

Comment:
The names here are bit different from the KEP: https://github.com/kubernetes/enhancements/pull/1234/files#diff-fd841be6aa8e9bedba8f79659f684798R455  Is that by design?

Reply:
Oh, I missed it! Thank you!
Will fix it.

Reply:
Fixed.